### PR TITLE
Solve an issue spot by the Static Analyzer in PixelCPETemplateReco.cc

### PR DESCRIPTION
**Pull request by perrotta**
I could not resist and I fixed an issue ("Value stored to 'distance_min' is never read" in corrispondence of previous L402 of RecoLocalTracker/SiPixelRecHits/src/PixelCPETemplateReco.cc), also profiting for simplifying a bit the code and avoiding useless calculation of a few square roots.

I only looked at that block, and I did not intervene in the overall logic of the code: nothing must change in the outputs.

This should not interfere with #20904, but to be sure I will not sign it until that one is merged in 94X

@ahinzmann

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: RecoLocalTracker/SiPixelRecHits/src/PixelCPETemplateReco.cc [hunk 1]
```diff
@@ -439,9 +439,11 @@ PixelCPETemplateReco::localError(DetParam const & theDetParam,  ClusterParam & t
    //cout << "CPETemplate : " << endl;
    
    //--- Default is the maximum error used for edge clusters.
-   const float sig12 = 1./sqrt(12.0);
-   float xerr = theDetParam.thePitchX *sig12;
-   float yerr = theDetParam.thePitchY *sig12;
+   //--- (never used, in fact: let comment it out, shut up the complains of the static analyzer, and save a few CPU cycles)
```
Comment (VinInn) [inline RecoLocalTracker/SiPixelRecHits/src/PixelCPETemplateReco.cc]: I bet the compiler optimizes out already...

Comment (perrotta) [reply RecoLocalTracker/SiPixelRecHits/src/PixelCPETemplateReco.cc]: Not the two products, I guess...
Anyhow, at least we can shut up a couple of warnings in the static analyzer (and concentrate on other possibly more interesting ones)

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Just some cleaning and simplification of the current code of PixelCPETemplateReco
- It allows fixing a few warning messages reported by the static analyzer
- No effect on reco output, none expected

**Comment by davidlange6** [comment]
+1

**Comment by pmaksim1** [comment]
---Hi folks, just catching up with this PR as well. This is fine to integrate.  The code contains in various places remnants of the original ORCA code I put in CMSSW in 2005.  So clearly there is a lot of innocent but useless junk like this lying around, covered by dust and cobwebs... 

For instance, I think that the code that was commented out came from the fact that, once upon a time, we were naive and thought that if we have a cluster with a single pixel (in projection), then the error is 1/sqrt(12).  But of course there is Lorentz drift, so the error is smaller (since the hit has to be "just right" so that everything drifts perfectly), but it is also more complicated since there would be charges in the neighboring pixels but below the threshold, etc.

**Pull request merged**